http/file_server_test.ts

```diff
@@ -498,3 +503,296 @@ Deno.test("file_server should show .. if it makes sense", async function (): Pro
+      await res.text();
+      assertEquals(res.status, 416);
+      assertEquals(res.statusText, "Range Not Satisfiable")
+      
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server should return 416 due to a bad range request (a-b)",
+  async () => {
+    await startFileServer();
+    try {
+      const headers = {
+        "range": "bytes=a-b"
+      }
+      const res = await fetch("http://localhost:4507/testdata/test%20file.txt", { headers });
+      await res.text();
+      assertEquals(res.status, 416);
+      assertEquals(res.statusText, "Range Not Satisfiable")
+      
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server returns correct mime-types",
+  async () => {
+    await startFileServer();
+    try {
+      const txtRes = await fetch("http://localhost:4507/testdata/test%20file.txt");
+      assertEquals(txtRes.headers.get("content-type"), "text/plain");
+      await txtRes.text(); // Consuming the body so that the test doesn't leak resources
+      
+      const htmlRes = await fetch("http://localhost:4507/testdata/hello.html");
+      assertEquals(htmlRes.headers.get("content-type"), "text/html");
+      await htmlRes.text(); // Consuming the body so that the test doesn't leak resources
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server sets `accept-ranges` header to `bytes` for directory listings",
+  async () => {
+    await startFileServer();
+    try {
+      const res = await fetch("http://localhost:4507/");
+      assertEquals(res.headers.get("accept-ranges"), "bytes");
+      await res.text(); // Consuming the body so that the test doesn't leak resources
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server sets `accept-ranges` header to `bytes` for file responses",
+  async () => {
+    await startFileServer();
+    try {
+      const res = await fetch("http://localhost:4507/testdata/test%20file.txt");
+      assertEquals(res.headers.get("accept-ranges"), "bytes");
+      await res.text(); // Consuming the body so that the test doesn't leak resources
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server sets `content-length` header correctly",
+  async () => {
+    await startFileServer();
+    try {
+      const res = await fetch("http://localhost:4507/testdata/test%20file.txt");
+      assertEquals(res.headers.get("content-length"), "10068");
+      await res.text(); // Consuming the body so that the test doesn't leak resources
+    } finally {
+      await killFileServer();
+    }
+  },
+);
+
+Deno.test(
+  "file_server sets `Last-Modified` header correctly",
+  async () => {
+    await startFileServer();
+    try {
+      const res = await fetch("http://localhost:4507/testdata/test%20file.txt");
+      assertEquals(res.headers.get("last-modified"), "Mon, 12 Jul 2021 23:31:46 GMT");
```

Comment:
git doesn't store mtime. ref https://softwareengineering.stackexchange.com/questions/350403/why-doesnt-git-set-the-file-time

You might need to get mtime from the actual file.

The same problem also seems affecting etag and if-none-match assertions

Reply:
I have changed it to pull the values dynamically at runtime. They all pass on my machine. We'll see about the CI.